src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -63,38 +64,45 @@
 Json = Union[Dict[str, Any], List[Any], str, int, float, bool, None]
 
 
-class RestZenStore(BaseZenStore):
-    """ZenStore implementation for accessing data from a REST API."""
+class RestZenStoreConfiguration(StoreConfiguration):
+    """REST ZenML store configuration.
 
-    def initialize(
-        self,
-        url: str,
-        *args: Any,
-        **kwargs: Any,
-    ) -> "RestZenStore":
-        """Initializes a rest zen store instance.
+    Attributes:
+        username: The username to use to connect to the Zen server.
+        password: The password to use to connect to the Zen server.
+    """
 
-        Args:
-            url: Endpoint URL of the service for zen storage.
-            args: additional positional arguments (ignored).
-            kwargs: additional keyword arguments (ignored).
+    type: StoreType = StoreType.REST
+    username: str
+    password: str = ""
 
-        Returns:
-            The initialized zen store instance.
+    class Config:
+        """Pydantic configuration class."""
 
-        Raises:
-            ValueError: if the url is not a valid URL.
-        """
-        if not self.is_valid_url(url.strip("/")):
-            raise ValueError("Invalid URL for REST store: {url}")
-        self._url = url.strip("/")
-        super().initialize(url, *args, **kwargs)
-        return self
+        # Validate attributes when assigning them. We need to set this in order
+        # to have a mix of mutable and immutable attributes
+        validate_assignment = True
```

Comment:
Any reason why extras should be ignored here? And the validate assignment is already set on the superclass and can probably be removed?

Reply:
I'm ignoring extras in all configuration related models persisted to disk as a general strategy to ensure backwards compatibility with previous ZenML versions.

Reply:
Mhm I see, I think for `StackComponent`s we recently did the opposite, we might want to unify that. I think in that case it was to prevent typos on the CLI though that parses arbitrary arguments, I'm guessing that's not a problem here right, or could something like this happen?
```bash
zenml store register --passwor=...
```

Reply:
No, the CLI is highly regulated concerning what users can configure for the stores. In fact, the local SQLite store is completely hidden through `zenml config set --local-store` and accepts no user parameters. Creating a REST store through the CLI is also tightly regulated through the use of specified `--url`, `--username` and  `--password` arguments.